ECHO-COVID19.ipynb

```diff
@@ -46,8 +65,195 @@
+    "# = total number of pollutants stack test. May want to summarize by facility instead. \n",
+    "stacks = stacks.resample('M').sum() #resample to a monthly basis  \n",
+    "stacks = stacks.loc[(stacks.index.month == 4)] # Filter back to just Aprils\n",
+    "stacks = pd.DataFrame(stacks)\n",
+    "\n",
+    "ax = stacks.plot(kind='bar', title = \"Past Aprils - # of pollutants stack-tested\", figsize=(20, 10), fontsize=16)\n",
+    "ax\n",
+    "\n",
+    "#label trendline\n",
+    "trend=stacks['STATE_EPA_FLAG'].mean()\n",
+    "ax.axhline(y=trend, color='r', linestyle='-')\n",
+    "\n",
+    "#label past 4 Aprils trend (2017, 2018, 2019, 2020)\n",
+    "trend_month=pd.concat([stacks.loc[\"2017-04\"],stacks.loc[\"2018-04\"],stacks.loc[\"2019-04\"]])\n",
+    "trend_month=trend_month['STATE_EPA_FLAG'].mean()\n",
+    "ax.axhline(y=trend_month, color='y', linestyle='-')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### What are facilities releasing in to the air?\n",
+    "Ramifications for COVID-19"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = \"select * from `ICIS-AIR_VIOLATION_HISTORY` where HPV_DAYZERO_DATE like '04-%'\"\n",
+    "url='http://apps.tlt.stonybrook.edu/echoepa/?query='\n",
+    "data_location=url+urllib.parse.quote(sql)\n",
+    "\n",
+    "data = pd.read_csv(data_location,encoding='iso-8859-1',header = 0)\n",
+    "data.set_index(\"pgm_sys_id\", inplace=True)\n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data['HPV_DAYZERO_DATE'] = pd.to_datetime(data['HPV_DAYZERO_DATE'], format=\"%m-%d-%Y\") #format\n",
+    "\n",
+    "airviols = data.groupby(['HPV_DAYZERO_DATE'])['ENF_RESPONSE_POLICY_CODE'].count() \n",
+    "# = total number of violations. May want to summarize by facility instead. \n",
+    "airviols = airviols.resample('M').sum() #resample to a monthly basis  \n",
+    "airviols = airviols.loc[(airviols.index.month == 4)] # Filter back to just Aprils\n",
+    "airviols = pd.DataFrame(airviols)\n",
+    "\n",
+    "ax = stacks.plot(kind='bar', title = \"Past Aprils - # of Air Violations\", figsize=(20, 10), fontsize=16)\n",
+    "ax\n",
+    "\n",
+    "#label trendline\n",
+    "trend=airviols['ENF_RESPONSE_POLICY_CODE'].mean()\n",
+    "ax.axhline(y=trend, color='r', linestyle='-')\n",
+    "\n",
+    "#label past 4 Aprils trend (2017, 2018, 2019, 2020)\n",
+    "trend_month=pd.concat([airviols.loc[\"2017-04\"],airviols.loc[\"2018-04\"],airviols.loc[\"2019-04\"]])\n",
+    "trend_month=trend_month['ENF_RESPONSE_POLICY_CODE'].mean()\n",
+    "ax.axhline(y=trend_month, color='y', linestyle='-')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Water pollutant discharges"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
This cell is taking a really long time! Might want a note to warn

Reply:
Actually it took so long that I stopped it and skipped this section

Reply:
uhoh! I can't see what cell it is though...

Reply:
Yes it took a while for me too. Added a note about this and also will think of ways to make it quicker....

Reply:
I thought only loading data for Aprils would make it manageable! Imagining loading ALL of it (~1GB)

Reply:
I think there was a line commented out in the following cell that would have caused an error too. Fixed.